src/core/core.scale.js

```diff
@@ -193,8 +193,11 @@ module.exports = function(Chart) {
 			var tickLabelFont = helpers.fontString(tickFontSize, tickFontStyle, tickFontFamily);
 			context.font = tickLabelFont;
 
-			var firstWidth = context.measureText(me.ticks[0]).width;
-			var lastWidth = context.measureText(me.ticks[me.ticks.length - 1]).width;
+			var firstTick = me.ticks[0];
+			var firstWidth = helpers.isArray(firstTick) ? helpers.longestText(context, tickLabelFont, firstTick) : context.measureText(firstTick).width;
```

Comment:
Yup, this needs a lot more cleaning. Once the other 2 PRs ahead of this are merged, I'll come back to this one otherwise there will be too many conflicts